src/main/java/io/strimzi/kafka/bridge/SinkBridgeEndpoint.java

```diff
@@ -225,61 +226,89 @@ protected void assign(boolean shouldAttachHandler) {
-                    return;
+            // fill a list with all available partitions as result of the partitionsFor for all topic subscriptions
+            List<PartitionInfo> availablePartitions = new ArrayList<>();
+            for (int i = 0; i < partitionsForHandlers.size(); i++) {
+                // check if, for each future, the partitionsFor operation is completed successfully or failed
+                if (partitionsResult.result() != null && partitionsResult.result().succeeded(i)) {
+                    availablePartitions.addAll(partitionsResult.result().resultAt(i));
+                } else {
+                    // TODO: what to do?
+                    //  it seems cannot be failed ones. The native Kafka client doesn't raise exceptions
+                    //  for not existing partitions
                 }
+            }
+
+            for (SinkTopicSubscription topicSubscription : this.topicSubscriptions) {
 
-                log.debug("Assigned to {} partition {}", this.topicSubscription().getTopic(), this.topicSubscription().getPartition());
-                // start reading from specified offset inside partition
-                if (this.topicSubscription().getOffset() != null) {
+                // check if a requested partition for a topic exists in the available partitions for that topic
+                Optional<PartitionInfo> requestedPartitionInfo =
+                        availablePartitions.stream()
+                                .filter(p -> p.getTopic().equals(topicSubscription.getTopic()) &&
+                                            p.getPartition() == topicSubscription.getPartition())
+                                .findFirst();
 
-                    log.debug("Seeking to offset {}", this.topicSubscription().getOffset());
+                this.handlePartition(Future.succeededFuture(requestedPartitionInfo));
 
-                    this.consumer.seek(topicPartition, this.topicSubscription().getOffset(), seekResult -> {
+                if (requestedPartitionInfo.isPresent()) {
+                    log.debug("Requested partition {} for topic {} does exist",
+                            topicSubscription.getPartition(), topicSubscription.getTopic());
+                    TopicPartition topicPartition = new TopicPartition(topicSubscription.getTopic(), topicSubscription.getPartition());
 
-                        this.handleSeek(seekResult);
-                        if (seekResult.failed()) {
+                    this.consumer.assign(Collections.singleton(topicPartition), assignResult -> {
+
+                        this.handleAssign(assignResult);
+                        if (assignResult.failed()) {
                             return;
                         }
-                        partitionsAssigned(Collections.singleton(topicPartition));
+
+                        log.debug("Assigned to {} partition {} on topic {}", topicSubscription.getTopic(), topicSubscription.getPartition());
```

Comment:
I think the topic and partition are switched here. Also there are three "{}", what is the third parameter to print?